### PR TITLE
SFD-XXXX Fix fuzzy sort issue and queries refactor

### PR DESCRIPTION
**Pull request by rohith1122**

### CODE REVIEW

File: application/app/src/main/java/uk/nhs/digital/uec/api/service/impl/FuzzyServiceSearchService.java [hunk 1]
```diff
@@ -107,24 +91,21 @@ public List<DosService> retrieveServicesByGeoLocation(
...
+      else {
+        if (dosService.getGeoLocation().getLon() == 0D && dosService.getGeoLocation().getLat() == 0D) {
           nonPopulatedLatLongServices.add(dosService);
         }
-        dosService.setDistance(locationService.distanceBetween(source, dosService.getLocation()));
       }
     }
 
-    this.populateServiceDistancesWithNorthingAndEastings(nonPopulatedLatLongServices, searchPostcode);
+    //this.populateServiceDistancesWithNorthingAndEastings(nonPopulatedLatLongServices, searchPostcode);
```
Comment (daonhs) [inline application/app/src/main/java/uk/nhs/digital/uec/api/service/impl/FuzzyServiceSearchService.java:104]: Remove the commented code if we don't need it any more

File: application/app/src/main/java/uk/nhs/digital/uec/api/service/impl/FuzzyServiceSearchService.java [hunk 2]
```diff
@@ -134,71 +115,71 @@ public List<DosService> retrieveServicesByGeoLocation(
...
-    List<String> postCodes =
-      dosServices.stream()
-        .filter(t -> t.getEasting() == null && t.getNorthing() == null)
-        .map(DosService::getPostcode)
-        .collect(Collectors.toList());
-    return !CollectionUtils.isNullOrEmpty(postCodes)
-      ? locationService.getLocationsForPostcodes(postCodes, headers)
-      : Collections.emptyList();
-  }
+//  private List<DosService> populateServiceDistancesWithNorthingAndEastings(List<DosService> nonPopulatedLatLongServices, String searchPostcode) throws InvalidParameterException, NotFoundException {
```
Comment (daonhs) [inline application/app/src/main/java/uk/nhs/digital/uec/api/service/impl/FuzzyServiceSearchService.java:118]: I would suggest implementing the google api to get the lat and long from the user request so we can delete this logic

File: application/app/src/test/java/uk/nhs/digital/uec/api/repository/ServiceRepositoryTest.java [hunk 3]
```diff
@@ -1,46 +1,46 @@
...
-import static org.mockito.ArgumentMatchers.anyInt;
-import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.List;
+import java.util.*;
```
Comment (daonhs) [inline application/app/src/test/java/uk/nhs/digital/uec/api/repository/ServiceRepositoryTest.java:10]: Import only what we need

**Review by daonhs — CHANGES_REQUESTED** [review]

File: application/app/src/main/java/uk/nhs/digital/uec/api/service/impl/FuzzyServiceSearchService.java (hunk 1 above)
Comment (rohith1122) [reply application/app/src/main/java/uk/nhs/digital/uec/api/service/impl/FuzzyServiceSearchService.java:104]: Sure , i will remove

File: application/app/src/test/java/uk/nhs/digital/uec/api/repository/ServiceRepositoryTest.java (hunk 3 above)
Comment (rohith1122) [reply application/app/src/test/java/uk/nhs/digital/uec/api/repository/ServiceRepositoryTest.java:10]: Sure , i will update

File: application/app/src/main/java/uk/nhs/digital/uec/api/service/impl/FuzzyServiceSearchService.java (hunk 2 above)
Comment (rohith1122) [reply application/app/src/main/java/uk/nhs/digital/uec/api/service/impl/FuzzyServiceSearchService.java:118]: I see where you are coming from with regard to empty locations. I will integrate google API

**Comment by rohith1122** [comment]
renamed SFD-XXXX to  SFD-4947

**Pull request closed without merging**